Add validator target to makefile

abought opened the pull request:
Allow people to run the validator from the same Makefile that controls all other tasks.  (As inspired by discussion with @tbekolay )

The new command is `make check`. It is equivalent to:
`python tools/check.py .`.

By design, the validator exits with error code `1` when validation fails. Can this be suppressed when run through `make`? Currently, validation failure results in the following ugly message when the above command is used:

`make: *** [check] Error 1`

abought committed:
Add validator target to makefile

wking commented:
On Mon, Feb 09, 2015 at 08:37:20PM -0800, Andy Boughton wrote:

> By design, the validator exits with error code `1` when validation
> fails. Can this be suppressed when run through `make`? Currently,
> validation failure results in the following ugly message when the
> above command is used:
> 
> `make: *** [check] Error 1`

That's not a particularly ugly message, and having Make explain that a
command failed and which command it was is also by design ;).  I'd
just embrace it.  If it really bugs you, POSIX specifies an -i option
that tells make to ignore error codes [1](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/make.html), so you can run `make -i
check` locally.

tbekolay commented:
Thanks, I was going to make a similar PR! :rainbow: Early on, I had the impression that we wouldn't be using Makefiles; since we switched to using them, I assumed that anything I would have to do in a lesson repository would be shown with `make commands`. So, I'm a strong +1 on this PR to prevent this confusion for other people writing lessons. I might also suggest adding a note to the `make unittest` target, but if `make check` is there I don't think people will be as tempted to run it.

gvwilson commented:
We want to keep Make out of the workshop-template workflow, but it's 
fine to use for lesson-template.

rgaiacs commented on Makefile:
```diff
@@ -53,6 +53,10 @@ unittest: tools/check.py tools/validation_helpers.py tools/test_check.py
 	cd tools/ && python2 test_check.py
 	cd tools/ && python3 test_check.py
 
+# Template validation:
```
This line should be

```
## check  : Validate lessons against template
```

rgaiacs commented:
@abought :+1: to merge after you address my comment. It will allow to people get the target when run

```
$ make
```

If you could add one warning at `unittest` will be awesome.

abought committed:
Improve messages for makefile targets

abought commented:
Messages added. Thanks for the feedback- I hadn't noticed the list of commands.

rgaiacs commented:
@abought Do you mind move `check` to before `unittest`? This will help people. =)

abought committed:
Reorder makefile commands.

First should come the common workflow tasks, then commands for internal/debugging use.

abought commented:
Good idea, @r-gaia-cs . While I was at it, I reordered a few more commands so the listing better matches the expected workflow/usage. Once we're confident this doesn't break anything, I'll let you make the final decision on merge.

Here is the new proposed order.

```
 $ make commands
 check    : Validate all lesson content against the template.
 clean    : Clean up temporary and intermediate files.
 preview  : Build website locally for checking.
 commands : Display available commands.
 settings : Show variables and settings.
 unittest : Run internal tests to ensure the validator is working correctly (for Python 2 and 3).
```

For comparison, here was the old order:

```
 $ make commands
 preview  : Build website locally for checking.
 unittest : Run internal tests to ensure the validator is working correctly (for Python 2 and 3)
 check    : Validate all lesson content against the template
 commands : Display available commands.
 settings : Show variables and settings.
 clean    : Clean up temporary and intermediate files.
```
